authors/apps/authentication/views.py

```diff
@@ -356,3 +363,57 @@ def post(self, request, **kwargs):
             }, status=status.HTTP_400_BAD_REQUEST)
         twitter_social_authentication = SocialLoginSignUp()
         return twitter_social_authentication.social_signup(user_info, **kwargs)
+
+
+class ReadingStats(RetrieveAPIView):
+    """
+    View class user reading statistics
+    """
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ReadingStatsSerializer
+
+    def ratio(self, views, reads):
+        try:
+            ratio = reads/views*100
+            return ratio
+        except:
+            return 0
+
+    def stats_in_last_30_days(self, obj, today, days, viewed, stats_for):
+        days = timedelta(hours=days)
+        begin = today - days
+        if stats_for == "views":
+            return obj.objects.filter(viewed_on__range=(begin, today), article=viewed)
+        elif stats_for == "reads":
+            return obj.objects.filter(read_on__range=(begin, today), article=viewed)
+        elif stats_for == "likes":
+            return obj.objects.filter(date_liked__range=(begin, today), article=viewed,
+                                      like_or_dislike=1)
+        elif stats_for == "dislikes":
+            return obj.objects.filter(date_liked__range=(begin, today), article=viewed,
+                                      like_or_dislike= -1)
+
+    def get(self, request, *args, **kwargs):
+        user = request.user
+        # compute article statistics
+        article_list = []
+        user_articles = Article.objects.filter(author=user)
+        # get views for each article
+        for user_article in user_articles:
+            viewed_articles = UserViews.objects.filter(article=user_article.id)
+            article_list.append({'article_id': user_article.id,
+                                 'article_title': user_article.title,
+                                 'total_views': len(UserViews.objects.filter(article=user_article.id)),
```

Comment:
Find a way to clean this up to PEP8 standards.

Reply:
I have refactored this method to match PEP8 standards. Thank you for pointing this out.